File: app/src/main/java/kashyap/genovatest/adaps/SeasonalListHorizBigAdap.java
package kashyap.genovatest.adaps;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import kashyap.genovatest.R;
import kashyap.genovatest.cusfo.ProximaNovaText;
import kashyap.genovatest.model.HomeModel;

/**
 * Created by VirtualDusk on 3/19/2018.
 */

public class SeasonalListHorizBigAdap extends RecyclerView.Adapter<SeasonalListHorizBigAdap.SingleItemRowHolder> {

    private ArrayList<HomeModel> itemsList;
    private Context mContext;
    private FragmentManager fm;

    SeasonalListHorizBigAdap(Context context, ArrayList<HomeModel> itemsList, FragmentManager fm) {
        this.itemsList = itemsList;
        this.mContext = context;
        this.fm = fm;
    }
    @Override
    public SingleItemRowHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(mContext).inflate(R.layout.adapter_seasonal, viewGroup, false);
        return new SingleItemRowHolder(v);

    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(final SingleItemRowHolder holder, @SuppressLint("RecyclerView") int i) {

       // holder.product_decsription.setText(itemsList.get(i).name);
    }

    @Override
    public int getItemViewType(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return itemsList.size();
    }

    public class SingleItemRowHolder extends RecyclerView.ViewHolder {

        ProximaNovaText product_decsription;


        public SingleItemRowHolder(View view) {
            super(view);
        //    this.product_decsription=  view.findViewById(R.id.product_desc);

        }

    }

}

File: app/src/main/java/kashyap/genovatest/adaps/SeasonalListHorizBigAdap.kt
package kashyap.genovatest.adaps

import android.annotation.SuppressLint
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.FragmentManager
import androidx.recyclerview.widget.RecyclerView
import kashyap.genovatest.R
import kashyap.genovatest.cusfo.ProximaNovaText
import kashyap.genovatest.model.HomeModel
import java.util.*

/**
 * Created by VirtualDusk on 3/19/2018.
 */
class SeasonalListHorizBigAdap internal constructor(private val mContext: Context, private val itemsList: ArrayList<HomeModel>, private val fm: FragmentManager) : RecyclerView.Adapter<SeasonalListHorizBigAdap.SingleItemRowHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SingleItemRowHolder {
        val v = LayoutInflater.from(mContext).inflate(R.layout.adapter_seasonal, viewGroup, false)
        return SingleItemRowHolder(v)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: SingleItemRowHolder, @SuppressLint("RecyclerView") i: Int) {

        // holder.product_decsription.setText(itemsList.get(i).name);
    }

    override fun getItemViewType(position: Int): Int {
        return position
    }

    override fun getItemCount(): Int {
        return itemsList.size
    }

    inner class SingleItemRowHolder(view: View?) : RecyclerView.ViewHolder(view!!) {
        var product_decsription: ProximaNovaText? = null
    }

}